distributions/LibreELEC/filesystem/usr/bin/update-system

```diff
@@ -0,0 +1,308 @@
+        return s[len(prefix):]
+    return s
+
+
+def rchop(s, suffix):
+    '''Remove suffix from string.'''
+    if suffix and s.endswith(suffix):
+        return s[:-len(suffix)]
+    return s
+
+
+def parse_osrelease():
+    '''Read /etc/os-release for key variables.'''
+    def strip_quotes(s):
+        '''Remove quotes from start and end of string.'''
+        s = lchop(s, '\"')
+        s = rchop(s, '\"')
+        return s
+
+
+    if os.path.isfile('/etc/os-release'):
+        with open('/etc/os-release', 'r', encoding='utf-8') as data:
+            content = data.read()
+    else:
+        die('ERROR: Did not find /etc/os-release.')
+    for line in content.splitlines():
+        if line[0:15] == 'LIBREELEC_ARCH=':
+            architecture = strip_quotes(line.split('=')[1])
+        elif line[0:5] == 'NAME=':
+            distribution = strip_quotes(line.split('=')[1])
+        elif line[0:8] == 'VERSION=':
+            version = strip_quotes(line.split('=')[1])
+        elif line[0:11] == 'VERSION_ID=':
+            version_id = strip_quotes(line.split('=')[1])
+    return architecture, distribution, version, version_id
+
+
+# Fetch json file
+def fetch_update_json():
+    '''Downloads releases.json file and readies for parsing.'''
+    if ALT_JSON:
+        if os.path.exists(ALT_JSON):
+            releases_json = f'file://{ALT_JSON}'
+        elif ALT_JSON.startswith('http://') or ALT_JSON.startswith('https://'):
+            releases_json = ALT_JSON
+        else:
+            die(f'ERROR: Unable to locate {ALT_JSON}.')
+    else:
+        releases_json = UPDATE_DOWNLOAD_URL
+    data = urllib.request.urlopen(releases_json)
+    if data:
+        content = data.read()
+        json_data = content.decode('utf-8').strip()
+    if not json_data:
+        die(f'ERROR: Failed to load releases.json')
+    return json.loads(json_data)
+
+
+# Fetch download file
+def fetch_update_file(update_url):
+    '''Download update_url to a temporary directory. Copy to LOCAL_UPDATE_DIR when finished.'''
+    update_filename = update_url.split('/')[-1]
+    with tempfile.TemporaryDirectory() as update_temp_dir:
+        try:
+            with urllib.request.urlopen(update_url) as download, open(f'{update_temp_dir}/update.file', 'wb') as file_in_progress:
+                shutil.copyfileobj(download, file_in_progress)
+        except Exception as e:
+            print(e)
+            # delete partial download
+            if os.path.exists(f'{update_temp_dir}/update.file'):
+                print('ERROR: Download failure. Removing partially downloaded file.')
+                os.remove(f'{update_temp_dir}/update.file')
+        else:
+            print('Download finished.')
+            if os.path.exists(f'{update_temp_dir}/update.file'):
+                print(f'Copying {update_filename} to {LOCAL_UPDATE_DIR}')
+                shutil.copy2(f'{update_temp_dir}/update.file', f'{LOCAL_UPDATE_DIR}/{update_filename}')
```

Comment:
sorry no real py knowledge but isn't that working similar to bash and creates a `//` ?
`LOCAL_UPDATE_DIR = '/storage/.update/'`
`f'{LOCAL_UPDATE_DIR}/{update_filename}')`
-> `/storage/.update//{update_filename}`

Reply:
Yes. Should be corrected.